arangod/Aql/ExecutionNode.cpp

```diff
@@ -2755,6 +2808,12 @@ CostEstimate ReturnNode::estimateCost() const {
   return estimate;
 }
 
+AsyncPrefetchEligibility ReturnNode::canUseAsyncPrefetching() const noexcept {
+  // ReturnNodes could make use of async prefetching, but the gain
+  // is too little to justify it.
+  return AsyncPrefetchEligibility::kDisableForNode;
```

Comment:
Actually, for streaming cursors prefetching from the return node is probably exactly what we want.